app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class TargetsController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_target, only: [:show]
+
+      def index
+        @targets = Target.all
+      end
+
+      def show; end
+
+      def create
+        new_target
+        @target.save ? render_create_success : render_create_error
+      end
+
+      private
+
+      def set_target
+        @target = Target.find(target_params[:title])
```

Comment:
Why don't you find by `:id` ?

Also, use `@target ||= current_user.targets.find...` and that way you don't override the target if it was already set with the `||=`